### PR TITLE
Registration user

### PR DESCRIPTION
**Pull request by condef5**
Feature #8: Registration users

This pull request include the following:
- View registration for user
- View login for user
- Create user for registration
- Refactor layout in partials
- Display flash messages

### CODE REVIEW

File: models/User.rb [hunk 1]
```diff
@@ -28,6 +30,22 @@ def self.relation(measures)
...
 
+  def self.create(user)
+    users = self.read
+    users.push({
+      "name" => user["name"],
+      "email" => user["email"],
+      "gender" => user["gender"],
+      "password" => user["password"],
+      "measures" => []
+    })
```
Comment (KattyaCuevas) [inline models/User.rb]: Is this the same of
```ruby
users << user.merge({ measures: [] })
```
?

File: views/layout.erb [hunk 2]
```diff
@@ -11,60 +11,20 @@
...
-    <script
-      defer
-      src="https://use.fontawesome.com/releases/v5.0.10/js/all.js"
-    ></script>
   </head>
-
   <body>
-    <nav class="navbar has-shadow is-fixed-top">
+    <%= erb :'partials/nav' %>
+    <section class="section <%= defined?(hero) ? 'hero is-success' : '' %> " id="content">
```
Comment (KattyaCuevas) [inline views/layout.erb]: Why do you call to `defined?` method?

File: views/layout.erb [hunk 3]
```diff
@@ -11,60 +11,20 @@
...
-                  Add
-                </a>
-                <a class="navbar-item" href="/user">
-                  Show
-                </a>
-              </div>
-            </div>
-          </div>
-        </div>
+        <%= erb :'partials/message' %>
```
Comment (KattyaCuevas) [inline views/layout.erb]: I think it's the same `:'partials/message'`  and `'partials/message'`.

File: models/User.rb (hunk 1 above)
Comment (condef5) [reply models/User.rb]: Yes, thank you now it looks super cool

File: routes.rb [hunk 4]
```diff
@@ -1,8 +1,33 @@
...
+  if session[:user_email]
+    @current_user = User.find(session[:user_email]) 
+    erb :home
+  else
+    erb :register, :locals => { :hero => true }
+  end
+end
+
+post '/register' do
+  User::create(params)
```
[the remark this answers is not in the document]
Comment (condef5) [reply routes.rb]: Of course, 2 years with php left havoc :)

File: views/layout.erb (hunk 2 above)
Comment (condef5) [reply views/layout.erb]: For add `hero is-success` class  in views register and login.
That class add  background pretty in the views.

File: views/layout.erb (hunk 3 above)
Comment (condef5) [reply views/layout.erb]: It's not the same, display literally `partial/message` but not renderize the partial.

Comment (KattyaCuevas) [reply views/layout.erb]: could you try with `render "partial/message"`, please ?

File: views/layout.erb (hunk 2 above)
Comment (KattyaCuevas) [reply views/layout.erb]: Is the same of `'hero is-success' if hero` ?

**Pull request merged**